src/dlsproto/client/request/internal/GetRange.d

```diff
@@ -398,11 +398,35 @@ private scope class GetRangeHandler
 
     private /* scope */ class RecordStream
     {
-        /// The acquired buffer to store a batch of records.
-        private void[]* batch_buffer;
+        /// Double buffer structure
+        private struct DoubleBuffer
+        {
+            /// Acquired buffers to store batches of records. Reader always
+            /// appends to the back one and the RecordStream always uses the
+            /// front one.
+            private void[]* front_buffer;
+
+            /// ditto
+            private void[]* back_buffer;
+
+            /// Initializes the double buffer
+            void init (void[]* delegate() getVoidBuffer)
+            {
+                this.front_buffer = getVoidBuffer();
+                this.back_buffer = getVoidBuffer();
+            }
+
+            /// Swaps the front and back buffer
+            private void swap ()
+            {
+                auto tmp = this.front_buffer;
+                this.front_buffer = this.back_buffer;
+                this.back_buffer = tmp;
```

Comment:
`typeid(void[]*).swap(&this.front_buffer, &this.back_buffer)` :trollface: 

(just kidding)